src/lib/ExclusivityLib.sol

```diff
@@ -54,7 +55,10 @@ library ExclusivityLib {
     /// @dev if the order has no exclusivity, always returns true
     /// @dev if the order has active exclusivity and the current filler is the exclusive address, returns true
     /// @dev if the order has active exclusivity and the current filler is not the exclusive address, returns false
-    function hasFillingRights(address exclusive, uint256 exclusivityEndTime) internal view returns (bool) {
-        return exclusive == address(0) || block.timestamp > exclusivityEndTime || exclusive == msg.sender;
+    function hasFillingRights(address exclusive, uint256 exclusivityEnd, bool timeBased) internal view returns (bool) {
+        return exclusive == address(0) ||
+            (timeBased && block.timestamp > exclusivityEnd) ||
+            (!timeBased && block.number > exclusivityEnd) ||
```

Comment:
if doing the above can just pass in `block.timestamp` or `block.number` as as `exclusivityCheck` param or something